mantidimaging/gui/windows/operations/presenter.py

```diff
@@ -196,8 +197,27 @@ def _post_filter(self, updated_stacks, task):
             self.view.clear_notification_dialog()
             self.view.show_operation_completed(self.model.selected_filter.filter_name)
 
-    def _do_apply_filter(self, apply_to):
-        self.model.do_apply_filter(apply_to, partial(self._post_filter, apply_to))
+    def _do_apply_filter(self, apply_to, allow_180_degree=False, confirm_with_user_for_180degree=False):
+        confirmed_stacks = []
+        for stack in apply_to:
+            is_180_proj = self.is_a_proj180deg(stack)
+            if not is_180_proj:
+                confirmed_stacks.append(stack)
+                continue
+            if is_180_proj and allow_180_degree:
+                confirmed_stacks.append(stack)
+            else:
+                if is_180_proj and confirm_with_user_for_180degree and\
+                        self.view.ask_confirmation("Operations applied to the sample are also automatically applied to "
+                                                   "the 180 degree projection. Please avoid applying an operation "
+                                                   "unless you're absolutely certain you need to.\nAre you sure you "
+                                                   "want to apply to 180 degree projection?"):
+                    confirmed_stacks.append(stack)
```

Comment:
This bit and the similar part of code in `_post_filter` has become very hard to understand, can it be refactored into something more straightforward? On a glance it seems like we should be able to branch out based on whether the current stack is a 180deg projection one or not, but I haven't looked too much at it

Reply:
Also this whole asking code should be in `do_apply_filter` to avoid cluttering this function that should be remain simple and just do the running.

I think this can simplify it to essentially being `if is_180_proj and self.view.ask_confirmation(...)` in `do_apply_filter`, so if the user clicks OK it just runs the stack through the rest

Reply:
The code is in _do_apply_filter() due to the fact it is called in 3 places and I do not want to repeat the code. It makes sense to centralize it here, I will attempt to simplify the code as I agree it was not as clear as it could have been.

Reply:
I think we only care to show this message if the user is trying to apply directly on 1 stack, and that stack happens to be proj 180 degree. The only entry point for this case is `do_apply_filter` if I'm not mistaken